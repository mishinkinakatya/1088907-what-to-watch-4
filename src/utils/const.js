export const DEFAULT_GENRE = `All genres`;

export const TabsName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const ActionType = {
  CHANGE_ACTIVE_GENRE: `CHANGE_ACTIVE_GENRE`,
  CHANGE_MAX_COUNT_OF_VISIBLE_MOVIES: `CHANGE_MAX_COUNT_OF_VISIBLE_MOVIES`,
  CHANGE_STATUS_AUTHORIZATION_ERROR: `CHANGE_STATUS_AUTHORIZATION_ERROR`,
  LOAD_MOVIES: `LOAD_MOVIES`,
  LOAD_REVIEWS: `LOAD_REVIEWS`,
  LOAD_PROMO_MOVIE: `LOAD_PROMO_MOVIE`,
  LOAD_FAVORITE: `LOAD_FAVORITE`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  CHANGE_STATUS_OF_SENDING_REVIEW: `CHANGE_STATUS_OF_SENDING_REVIEW`,
  CHANGE_AUTH_INFO: `CHANGE_AUTH_INFO`,
  CHANGE_SEND_FAVORITE_STATUS: `CHANGE_SEND_FAVORITE_STATUS`,
  SHOW_LOAD_ERROR: `SHOW_LOAD_ERROR`,
};

export const INITIAL_COUNT_VISIBLE_MOVIES_ON_MAIN_PAGE = 8;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const SendingStatus = {
  NO_SENDING: `NO_SENDING`,
  SENDING: `SENDING`,
  SUCCESS: `SUCCESS`,
  FAIL: `FAIL`,
};

export const AppRoute = {
  MAIN_PAGE: `/`,
  SIGN_IN_PAGE: `/login`,
  MY_LIST_PAGE: `/mylist`,
  MOVIE_PAGE: `/films`,
  ADD_REVIEW_PAGE: `/review`,
  PLAYER_PAGE: `/player`,
};

export const AppPages = {
  MAIN_PAGE: `MAIN_PAGE`,
  SIGN_IN_PAGE: `SIGN_IN_PAGE`,
  MY_LIST_PAGE: `MY_LIST_PAGE`,
  MOVIE_PAGE: `MOVIE_PAGE`,
  ADD_REVIEW_PAGE: `ADD_REVIEW_PAGE`,
  ERROR_PAGE: `ERROR_PAGE`,
  LOADING_PAGE: `LOADING_PAGE`,
};
