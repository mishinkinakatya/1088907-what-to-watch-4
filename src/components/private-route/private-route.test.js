import React from "react";
import renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import PrivateRoute from "./private-route.jsx";
import NameSpace from "../../store/name-space.js";
import {AuthorizationStatus, AppRoute} from "../../utils/const.js";
import {Router} from 'react-router-dom';
import history from '../../history';


const mockStore = configureStore([]);

describe(`PrivateRoute`, () => {
  it(`Render PrivateRoute`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const tree = renderer
      .create(
          <Router history={history}>
            <Provider store={store}>
              <PrivateRoute
                path={AppRoute.MY_LIST_PAGE}
                exact={true}
                render={() => {}}
              />
            </Provider>
          </Router>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
