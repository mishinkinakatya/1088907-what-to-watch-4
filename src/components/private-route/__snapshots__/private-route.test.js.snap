// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Render PrivateRoute 1`] = `null`;
